### DISCUSSION TITLE
Very strange locking issue with forward slash /

### DISCUSSION THREAD

**Discussion by igray** [LazyVim/LazyVim#2112] [Q&A] [answered]
i am experiencing an issue using LazyVim where if I'm in insert mode and I type a / character, neovim freezes. If I wait long enough (like 3-5 minutes) it can come back, but the issues described later remain. If I type ./, the LSP kick in, browsing local files, everything is fine-- this only happens when I type / with no other context.

I'm using NixOS with Hyprland and Kitty. This started happening a few weeks ago, but I'm unsure of the exact update when it started. I'm currently at stable 23.11, but this was happening on stable 23.05 before I upgraded.

What makes this difficult is that I'm using Hyprland, and once this happens it renders my desktop mostly useless: most Hyprland hotkeys stop responding as well. For example, I can not start Fuzzel (app-launcher) to open a new app and I can not open a new terminal. I have an Eww app in my Waybar that I use to logout, reboot, etc and that stops responding. I can still use a hotkey to kill Hyprland, but doing so crashes greetd and I'm forced to reboot to get back to Hyprland. My theory is that once this happens it does something to the xdg-portal that Once this has happened, I must switch to a new console (Ctrl-Alt-F2), login and reboot. Even if I kill Hyprland, it crashes greetd, so I must reboot to recover.

I do not experience this with an unconfigured neovim. I tried a [fresh/vanilla LazyVim configuration](https://www.lazyvim.org/installation) and it happens. If I run Gnome, it happens but at least it does not crash my desktop-- I can just open a new terminal and go on my merry way. I've also tried other terminals: alacritty and gnome console and it happens with both.

The next thing I'm going to try is to disable each plug-in one-by-one to see what is causing it. I just wanted to post here to see if anyone had any better ideas.

I've searched for anyone else experiencing this issue, but it's hard to search for "/" and "slash" mostly talks about people having issues with Windows vs. Linux/OSX.

**Comment by dpetka2001** [comment, marked as answer]
The only thing I can think of when you press `/` in insert mode is the `path` source of `nvim-cmp`, which tries to populate a completion menu with filenames available. If you press `./` then it populates file paths from the directory of the buffer you're in downwards. You could try to disable the `path` source with something like 
```lua
  {
    "nvim-cmp",
    opts = function(_, opts)
      opts.sources = vim.tbl_filter(function(path)
        return not vim.tbl_contains({ "path" }, path.name)
      end, opts.sources)
    end,
  },
```
Test it to see if it solves your problem.

**Reply by igray** [reply]
Thank you so much @dpetka2001! That totally worked. I can live without path completion.

**Reply by dpetka2001** [reply]
You can still use `<C-x><C-f>` for Neovim's omnifunc path completion. It just won't be in a menu accessible through `nvim-cmp`'s auto-completion menu.

**Reply by gustavotr** [reply]
I am having the same problem but I would like to keep the auto-completion working. Any idea how to fix this?

**Reply by dpetka2001** [reply]
You can still use `<C-x><C-f>` Neovim's path auto-completion even if you disable the `path` source. Unfortunately, I can't think of another way. You should open an issue at the plugin's repo, so that the maintainer can assist you further.
